src/github.com/kelseyhightower/confd/backends/rancher/client.go

```diff
@@ -0,0 +1,135 @@
+package rancher
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	log "github.com/kelseyhightower/confd/log"
+)
+
+const (
+	MetaDataURL = "http://rancher-metadata/2015-07-25"
+)
+
+type Client struct {
+	Url        string
+	HttpClient *http.Client
+}
+
+func NewRancherClient(backendNodes []string) (*Client, error) {
+	url := MetaDataURL
+
+	if len(backendNodes) > 0 {
+		url = strings.Join([]string{"http://", backendNodes[0], "/2015-07-25"}, "")
+	}
+
+	log.Info(strings.Join([]string{"Using Rancher Metadata URL: ", url}, ""))
+	client := &Client{
+		Url:        url,
+		HttpClient: &http.Client{},
+	}
+
+	err := client.testConnection()
+	if err != nil {
+		return client, err
+	}
+
+	return client, nil
+}
+
+func (c *Client) GetValues(keys []string) (map[string]string, error) {
+	vars := make(map[string]string)
```

Comment:
for whatever reason this syntax is more popular

`vars := map[string]string{}`